File: src/main/java/thecodewarrior/logistic/api/Network.java
package thecodewarrior.logistic.api;

import com.teamwizardry.librarianlib.common.util.saving.Savable;
import com.teamwizardry.librarianlib.common.util.saving.Save;

import java.util.ArrayList;
import java.util.UUID;

/**
 * Created by TheCodeWarrior
 */
@Savable
public class Network {
	
	public Network() {
		
	}
	
	public int id = -1;
	
	@Save
	public ArrayList<UUID> nodes = new ArrayList<>();
}

File: src/main/java/thecodewarrior/logistic/api/impl/Network.kt
package thecodewarrior.logistic.api.impl

import thecodewarrior.logistic.api.IDrone
import thecodewarrior.logistic.api.INetwork
import java.util.*

/**
 * Created by TheCodeWarrior
 */
class Network(private val _id: Int) : INetwork {
    private val _drones = LinkedList<IDrone?>()
    private val _nodes = mutableListOf<UUID>()

    override fun getId() = _id
    override fun getNodes() = _nodes
    override fun getDrones() = _drones

}
